services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowHistorySpec.groovy

```diff
@@ -0,0 +1,125 @@
+        Long timestampAfterCreate = System.currentTimeMillis() / 1000L
+        checkHistoryCreateAction(flow.id, timestampAfterCreate)
+
+        when: "Update the created flow"
+        flowHelper.updateFlow(flow.id, flow.tap { it.description = it.description + "updated" })
+
+        then: "History record is created after updating the flow"
+        Long timestampAfterUpdate = System.currentTimeMillis() / 1000L
+        checkHistoryUpdateAction(flow.id, timestampAfterUpdate)
+
+        when: "Delete the updated flow"
+        flowHelper.deleteFlow(flow.id)
+
+        then: "History is still available for the deleted flow"
+        checkHistoryDeleteAction(flow.id, timestampAfterUpdate)
+    }
+
+    def "History records are created for the create/update actions using default timeline"() {
+        when: "Create a flow"
+        def (Switch srcSwitch, Switch dstSwitch) = topology.activeSwitches
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flowHelper.addFlow(flow)
+
+        then: "History record is created"
+        checkHistoryCreateAction(flow.id)
+
+        when: "Update the created flow"
+        flowHelper.updateFlow(flow.id, flow.tap { it.description = it.description + "updated" })
+
+        then: "History record is created after updating the flow"
+        checkHistoryUpdateAction(flow.id)
+
+        when: "Delete the updated flow"
+        flowHelper.deleteFlow(flow.id)
+
+        then: "History is still available for the deleted flow"
+        checkHistoryDeleteAction(flow.id)
+    }
+
+    def "History should not be returned in case timeline is incorrect (timeBefore > timeAfter)"() {
+        when: "Create a flow"
+        def (Switch srcSwitch, Switch dstSwitch) = topology.activeSwitches
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flowHelper.addFlow(flow)
+
+        then: "History record is created"
+        Long timestampAfterCreate = System.currentTimeMillis() / 1000L
+        checkHistoryCreateAction(flow.id, timestampAfterCreate)
+
+        when: "Try to get history for incorrect timeline"
+        def flowH = northbound.getFlowHistory(flow.id, timestampAfterCreate, timestampBefore)
+
+        then: "History record is NOT returned"
+        flowH.isEmpty()
+
+        and: "cleanup: Restore default state(remove created flow)"
+        flowHelper.deleteFlow(flow.id)
+    }
+
+    def "History should not be returned in case flowId was never created"() {
+        when: "Try to get history for incorrect flowId"
+        def flowHistory = northbound.getFlowHistory(NON_EXISTENT_FLOW_ID)
+
+        then: "History record is NOT returned"
+        flowHistory.isEmpty()
+    }
+
+    def checkHistoryCreateAction(flowId, timestampAfterCreate = null) {
+        def timeBefore = timestampAfterCreate ? timestampBefore : null
+        def flowHistory = northbound.getFlowHistory(flowId, timeBefore, timestampAfterCreate)
+
+        flowHistory.size() == 1
```

Comment:
Need to add `assert` everywhere. Groovy truth can be used only in `then/and` block directly.